### PR TITLE
Update @type/enzyme to include hostNodes()

### LINKED ISSUE #20288: @types/enzyme - Missing type definition for hostNodes()
- [ ] I tried using the `@types/enzyme` package and had problems.
- [ ] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [ ] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @MarianPalkus @NoHomey @jwbay @huhuanming @MartynasZilinskas @thovden  
--- 

Enzyme 3.1.0 introduced the hostNodes() method for ShallowWrapper and ReactWrapper which is not present in the type definition

### PR DESCRIPTION
**Pull request by mergepls**
Fixes #20288

Please fill in this template.

- [ ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/airbnb/enzyme/pull/1179>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/enzyme/index.d.ts [hunk 1]
```diff
@@ -430,7 +430,15 @@ export interface ShallowWrapper<P = {}, S = {}> extends CommonWrapper<P, S> {
      * NOTE: can only be called on wrapper of a single non-DOM component element node.
      * @param [options]
      */
-    dive<P2, S2>(options?: ShallowRendererProps): ShallowWrapper<P2, S2>;
+    dive<P2, S2>(options?: ShallowRendererProps): ShallowWrapper<any, any>;
```
Comment (jwbay) [inline types/enzyme/index.d.ts]: Don't believe this line needs to be changed

File: types/enzyme/index.d.ts [hunk 2]
```diff
@@ -430,7 +430,15 @@ export interface ShallowWrapper<P = {}, S = {}> extends CommonWrapper<P, S> {
...
+
+    /**
+     * Strips out all the not host-nodes from the list of nodes
+     *
+     * This method is useful if you want to check for the presence of host nodes
+     * (actually rendered HTML elements) ignoring the React nodes.
+     */
+    hostNodes(): ShallowWrapper<any, any>;
```
Comment (jwbay) [inline types/enzyme/index.d.ts]: These can be `Shallow|ReactWrapper<HTMLAttributes>` for better completions on calls chained after this

**Review by jwbay — CHANGES_REQUESTED** [review]

File: types/enzyme/index.d.ts (hunk 2 above)
Comment (ericlam51) [reply types/enzyme/index.d.ts]: this method is in the ShallowWrapper interface, therefore should it only be `ShallowWrapper<HTMLAttributes, any>`?

File: types/enzyme/index.d.ts (hunk 1 above)
Comment (ericlam51) [reply types/enzyme/index.d.ts]: indeed it does not need to

File: types/enzyme/index.d.ts (hunk 2 above)
Comment (jwbay) [reply types/enzyme/index.d.ts]: Right, yeah, was just being lazy, sorry. `ShallowWrapper<HTMLAttributes>` here and `ReactWrapper<HTMLAttributes>` below. The reason for omitting any is that the second parameter is state, and host nodes can't have state. So the default of `{}` handles shutting down access to `wrapper.state().x`.

Comment (ericlam51) [reply types/enzyme/index.d.ts]: Ah alright, thanks for the info. will push the changes

**Review by jwbay — APPROVED** [review]

**Comment by ericlam51** [comment]
@jwbay can you reapprove the PR, maybe it will remove the "Revision needed" label

**Review by jwbay — APPROVED** [review]

**Comment by ericlam51** [comment]
@RyanCavanaugh typescript-bot is misbehaving, I did the changes mentioned by jwbay but it still says "Revision needed"

**Comment by RyanCavanaugh** [comment]
The bot works kinda weird - it fetches data for all PRs, processes them, then applies all the actions. So it downloaded the "Changes requested" state, decided to mark Revision Needed, then jwbay approved, then the bot added the label. Then on the next iteration it saw the approve.

**Pull request merged**